### DISCUSSION TITLE
Podman binary size

### DISCUSSION THREAD

**Discussion by niklasHochst** [containers/podman#23049] [Q&A] [answered]
Hi, 

does anyone know why binaries I build locally are much bigger than the release binaries? Where can I find the build tags used for releases, or is this also dependent on environment variables, I tried setting the same environment variables as in cirrus.yml but this didn't change anything.

Thank you for helping me!

**Comment by Luap99** [comment]
Local builds have have a lot of debug symbols in the binary. Package builds generally remove the debug info (often place it in a separate package).
You can run `strip bin/podman` top remove them from your build binary which should get you close to the released version.

**Reply by niklasHochst** [reply]
Thank you very much! 
This already helps me a lot, since this will be used in embedded Devices I would still need to reduce the size. Do you know other options which could decrease the size by any chance? Sorry for my bad Go and Podman knowledge

**Reply by Luap99** [reply, marked as answer]
Unfortunately go binaries are very large. Podman is big application with all the features it has. I am not aware of any other tricks to drastically decrease its size.

Overall there is much that could be done within the podman code itself. Analyse the code and find out big packages and try to remove/replace them with smaller alternatives. I have done so many times in the past but the rate of new code is high and most easy to fix binary size problems have been addresses so doing this work is difficult.

If you only target a specific use case you could try to just delete all code paths you don't care about but of course this requires deep knowledge of podman and even then would not really be maintainable as you have to rebase this all the time for new releases.

**Reply by niklasHochst** [reply]
Thank you!
